### DISCUSSION TITLE
How to remove user invitations

### DISCUSSION THREAD

**Discussion by luigiPinnaAevo** [Budibase/budibase#13078] [Help] [answered]
Hello,

I want to remove some user invitations but the system doesn't allow me to remove them. Follow an example about the situation:

![image](https://github.com/Budibase/budibase/assets/99037506/dd110380-4f46-4543-b8ff-8084a5a314a3)

Is there anyone who knows how to solve it?

Thank you

**Comment by ConorWebb96** [comment, marked as answer]
Hey @luigiPinnaAevo,

These expire after 7 days, so its likely that this has resolved itself. If you are on cloud you don't really have any other option that waiting the 7 days for these to expire.

If you are self-hosted you have other options, if you follow the below steps you should be able to remove these quickly.

1. Find the relevant Redis container name by looking for redis-service or similar in your list of running containers
2. Stop the Redis container
3. Remove the Redis container
4. Bring down all of the other containers and then start them up again - a fresh Redis container should be brought up with no pending user invites

I've broken down these steps further e.g. the commands you would want to use.

```bash
# Find the relevant Redis container name by looking for redis-service or similar in your list of running containers
docker ps

# Stop the Redis container
docker stop redis-service

# Remove the Redis container
docker rm redis-service

# Bring down all of the other containers and then start them up again - a fresh Redis container should be brought up with no pending user invites
docker-compose down
docker-compose up
```

Here is the [feature request ](https://github.com/Budibase/budibase/issues/12342) a different user raised. Feel free to comment on it to get it bumped in priority.

**Comment by luigiPinnaAevo** [comment]
Thank you @ConorWebb96

**Comment by PresidentCamach0** [comment]
Hello,

When you invite a user and they are sat in "Pending Invites" there are no options to make changes. 
For example:
- Cancel Invite
- Re-send Invite

I found this out when the email settings were not setup correctly - the email did not send and I cannot add the same email addresses again.

Thanks!

**Reply by ConorWebb96** [reply]
Feature request here for this: https://github.com/Budibase/budibase/issues/11044, if you are self-hosted you can perform my suggestion above and it will clear any pending invites. Otherwise you will have to wait the 7 day expiry time.

Hopefully this helps.
